### PR TITLE
[Test][Intro] Fix introspection test on API changes for Xcode 9 Beta 1

### PR DESCRIPTION
**Pull request by dalexsoto**
Bug filled for AVPlayerLooper. LoopingEnabled -> https://bugzilla.xamarin.com/show_bug.cgi?id=57302

Bug filled for CoreImage filters -> https://bugzilla.xamarin.com/show_bug.cgi?id=57350

Need to radar

- case "EKCalendarItem": // Fails with NSInvalidArgumentException +[EKCalendarItem frozenClass]: unrecognized selector sent to class, will fill a radar
- case "UITableViewFocusUpdateContext": // Objective-C exception thrown.  Name: NSInternalInconsistencyException Reason: Invalid parameter not satisfying: focusSystem, will fill a radar

### CODE REVIEW

File: src/foundation.cs [hunk 1]
```diff
@@ -12140,6 +12140,9 @@ interface NSDateInterval : NSCopying, NSSecureCoding {
 		bool ContainsDate (NSDate date);
 	}
 
+#if XAMCORE_4_0
+	[DisableDefaultCtor] // -init should never be called on NSUnit!
+#endif
```
Comment (spouliot) [inline src/foundation.cs]: It's better not to be under 4_0
And add manual, empty ctor in another files (all excluded in 4_0) along with an obsolete attribute

File: src/uikit.cs [hunk 2]
```diff
@@ -15839,6 +15839,9 @@ interface UIGuidedAccessRestrictionDelegate {
 		string GetDetailTextForGuidedAccessRestriction (string restrictionIdentifier);
 	}
 
+#if XAMCORE_4_0
+	[DisableDefaultCtor] // [Assert] -init is not a useful initializer for this class. Use one of the designated initializers instead
+#endif
```
Comment (spouliot) [inline src/uikit.cs]: Same

File: src/uikit.cs [hunk 3]
```diff
@@ -15908,6 +15911,9 @@ interface UIFocusItem : UIFocusEnvironment
 		bool CanBecomeFocused { get; }
 	}
 		
+#if XAMCORE_4_0
+	[DisableDefaultCtor] // [Assert] -init is not a useful initializer for this class. Use one of the designated initializers instead
+#endif
 	[NoWatch]
```
Comment (spouliot) [inline src/uikit.cs]: Same

File: src/foundation.cs (hunk 1 above)
Comment (dalexsoto) [inline src/foundation.cs]: Should we error? o just warn?

**Comment by dalexsoto** [comment]
@spouliot Done! Thanks for the feedback!

File: src/Foundation/NSUnit.cs [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+//
+
+using System;
+using XamCore.Foundation;
+using XamCore.ObjCRuntime;
+
+namespace XamCore.Foundation {
+#if !XAMCORE_4_0
+	public partial class NSUnit {
+		[Obsolete ("Use the overload instead")]
```
Comment (rolfbjarne) [inline src/Foundation/NSUnit.cs]: Use which overload?

**Comment by dalexsoto** [comment]
@rolfbjarne Fixed and bug filled here https://bugzilla.xamarin.com/show_bug.cgi?id=57350 also MtouchNoSymbolStrip fix is part of this PR

**Comment by dalexsoto** [comment]
Introspection is finally green 😀

**Pull request merged**